src/util/io_limiter.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub trait IOLimiter {
```

Comment:
why you extract a IOLimiter here? Do we really need this?

Reply:
There might be other I/O to limit in the future.

Reply:
so what other I/O? 

If you want to handle different I/O, you should use a limiter to handle io.Read or io.Write, but not an IOLimiter.